### PR TITLE
#484 check for notifications only if there are notification are present

### PR DESCRIPTION
**Pull request by neremic**
the cycle check for removing old notifications is now only performed if there are notifications that might be removed due to age

### CODE REVIEW

File: client/lib/yourturn/src/notification-bar/notification-bar.jsx [hunk 1]
```diff
@@ -24,8 +24,8 @@ Notification.displayName = 'Notification';
 // ========
 
 class NotificationBar extends React.Component {
-    constructor() {
-        super();
+    constructor(props) {
```
Comment (prayerslayer) [inline client/lib/yourturn/src/notification-bar/notification-bar.jsx]: Unnecessary change as you don't use `props` :)

Comment (neremic) [inline client/lib/yourturn/src/notification-bar/notification-bar.jsx]: Grmph :)

File: client/lib/yourturn/src/notification-bar/notification-bar.jsx [hunk 2]
```diff
@@ -34,17 +34,38 @@ class NotificationBar extends React.Component {
     dismiss(id) {
         this.props.dispatch(Actions.removeNotification(id));
     }
+    
+    componentWillReceiveProps(nextProps) {
+        const { notifications = [] } = nextProps;
+
+        if (!this.state.interval && notifications.length > 0) {
+            let interval = setInterval(() => {
```
Comment (prayerslayer) [inline client/lib/yourturn/src/notification-bar/notification-bar.jsx]: This code is duplicated with what's in `componentDidMount`. Can you refactor this in two functions `startCountdown` and `stopCountdown` or something?

**Comment by prayerslayer** [comment]
👍

**Comment by neremic** [comment]
:+1:
